### PR TITLE
fix: allow render function to return null

### PR DESCRIPTION
**Pull request by KnisterPeter**
In react components are allowed to return null from the render function

### CODE REVIEW

**Comment by developit** [comment]
Any TS folks like @mseddon want to approve?

**Comment by mseddon** [comment]
LGTM!

File: src/preact.d.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 declare namespace preact {
-	interface ComponentProps<C extends Component<any, any> | FunctionalComponent<any>> {
+	interface ComponentProps<C extends Component<PropsType, any> | FunctionalComponent<PropsType>, PropsType> {
```
Comment (UnwrittenFun) [inline src/preact.d.ts]: What's the reason for adding PropsType here?

Comment (KnisterPeter) [inline src/preact.d.ts]: Isn't it just more specific/correct than 'any'?

Comment (UnwrittenFun) [inline src/preact.d.ts]: As PropsType isn't used in this interface and isn't constrained in any way, it provides no more safety than simply any, and only adds complexity.

**Pull request merged**